cpp/src/arrow/record_batch.cc

```diff
@@ -664,7 +667,9 @@ Status RecordBatch::ValidateFull() const {
   return ValidateBatch(*this, /*full_validation=*/true);
 }
 
-const std::shared_ptr<Device::SyncEvent>& RecordBatch::GetSyncEvent() const { return nullptr; }
+const std::shared_ptr<Device::SyncEvent>& RecordBatch::GetSyncEvent() const {
+  return nullptr;
```

Comment:
For this case, you should have a `null_sync_event_` member variable because it's illegal to return a reference to a `std::shared_ptr<Device::SyncEvent>` created on the stack implicitly from `nullptr`.

Reply:
that's probably why i didn't have it returning a const reference in the first place haha. I'll make something static that I can return const refs to